haystack/pipeline_utils/indexing.py

```diff
@@ -128,7 +128,9 @@ def __init__(
         if "text/html" in supported_mime_types:
             from haystack.components.converters import HTMLToDocument
 
-            self.pipeline.add_component("html_file_converter", HTMLToDocument())
+            html_to_document_converter = HTMLToDocument(extractor_type="KeepEverythingExtractor")
```

Comment:
Would it make sense to have `"KeepEverythingExtractor"` be the default initialization? Seems as if `"DefaultExtractor"` may drop entire html text under certain conditions (haven't looked into it) which is unexpected.

Reply:
Here's the comments for the respective extractors from the underlying boilerpy3  lib:

class DefaultExtractor(Extractor):
    """
    Usually worse than ArticleExtractor, but simpler/no heuristics. A quite generic full-text extractor.
    """

class KeepEverythingExtractor(Extractor):
    """
    Marks everything as content. Dummy Extractor; should return the input text. Use this to double-check that your
    problem is within a particular BoilerpipeExtractor, or somewhere else.
    """

After switching to KeepEverythingExtractor I noticed the content returned from the RAG similarity seemed fine and the prompt seemed ok.

Reply:
So I don't think we should change the mappings to the underlying boilerpy3 lib, but we could see if ArticleExtractor gives results?

Reply:
This code is being removed so resolving this.
`Hey @hosermage we are removing this code, see https://github.com/deepset-ai/haystack/pull/6806 for reasons why`

[](https://discord.com/channels/993534733298450452/993539071815200889/1199002558560211039)